Formula/p/packetry.rb

```diff
@@ -0,0 +1,19 @@
+class Packetry < Formula
+  desc "Fast, intuitive USB 2.0 protocol analysis application for use with Cynthion"
+  homepage "https://github.com/greatscottgadgets/packetry"
+  url "https://github.com/greatscottgadgets/packetry/archive/refs/tags/v0.1.0.tar.gz"
+  sha256 "8d91ddc17883299f302752b12e11aa539c306304109d733c8863b7bc444c9629"
+  license "BSD-3-Clause"
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "gtk4"
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/packetry --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Packetry is a GUI application. I used `--version` because it is one of only two command-line operations available in this initial release and because the other one is expected to fail. Based on your feedback I added the other command-line operation even though it is expected to fail.

We may add more command-line operations (e.g. file operations) in future releases. If so, I will be happy add an improved test to the formula.